### ISSUE TITLE
Expression left untouched if can't fit in line length limit

### ISSUE THREAD

**Issue by YaLTeR** [rust-lang/rustfmt#2970]
`rustfmt 0.99.2-nightly (5c9a2b6 2018-08-07)`
Input like:
```rust
fn foo() {
    a.first().second().third().fourth(|| { let a = 1; }).fifth(some_argument).sixth("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
}
```
is left completely untouched due to being unable to fit the long argument, even in the presence of patterns that must be broken into multiple lines (`|| { let a = 1; }` in this case). I'd expect rustfmt to do the best it can even if that means leaving some lines overflowing (which would be better than not doing anything), for instance like this:
```rust
fn foo() {
    a.first()
        .second()
        .third()
        .fourth(|| {
            let a = 1;
        }).fifth(some_argument)
        .sixth(
            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
        );
}
```

**Comment by nrc** [comment]
That would be better, but I think this is p-low because it would be a fairly disruptive change and in any case the user has to fix this up. Also since the formatting causes an error I think we can fix this backwards compatibly post 1.0.

**Comment by calebcartwright** [comment]
I know this one was opened first, but closing in favor of #3863 given the additional discussions in that thread
